### DISCUSSION TITLE
Using transformer to get specific data values

### DISCUSSION THREAD

**Discussion by offpiste-reese** [Budibase/budibase#5377] [Help]
Calling a REST service and need to pull out a singular value from the response.  The actual response from the service is:
{"id":1,"result":"CIZzK_tZ~32rBKYy9CYteuaDVdvRIQEy","error":null}

When I set the call up in Budibase, the response shows as (this is what I get if I leave the transformer as return data):
{
  "value": "{\"id\":1,\"result\":\"Kevd4iw9lt5pGGDFMLaTUgciKm3WG022\",\"error\":null}"
}

I want to pull the result value out using a Transformer.  I have tried the following:
const resp = data
var sessionId = "12345"
sessionId = resp.result
return {"sessionId":sessionId}

I end up with an empty JSON object.  If I leave out sessionId = resp.result, I get a JSON object with sessionId set to 12345.  

Additionally, is there any way to put debugging statements into the Transformer?

**Comment by melohagan** [comment]
Hi @offpiste-reese 

Budibase is receiving a string of the JSON object. Try using `JSON.parse(data).result`

**Reply by offpiste-reese** [reply]
Thanks, that worked.  From the documentation, it appeared that data was a JSON object.

**Reply by melohagan** [reply]
Normally an api would return the JSON object. What is the content-type that appears under the headers?

![Screenshot 2022-04-13 at 09 00 12](https://user-images.githubusercontent.com/101575380/163128977-65a4505e-8ce3-4f99-a231-9860ee16db27.png)

**Reply by offpiste-reese** [reply]
I do have content-type set to application/json, so not sure why it is not returning a JSON object.

**Reply by melohagan** [reply]
Strange. Does it behave as expected if you hit your REST service in say postman.

**Reply by offpiste-reese** [reply]
Yes, I have called it from a Python script and Bubble and I don't have this issue.
